src/Symfony/Component/FeatureFlag/Provider/InMemoryProvider.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag\Provider;
+
+/**
+ * @experimental
+ */
+final class InMemoryProvider implements ProviderInterface
+{
+    /**
+     * @param array<string, (\Closure(): mixed)> $features
+     */
+    public function __construct(
+        private readonly array $features,
```

Comment:
There is no need to maintain code compatibility with PHP 8.1.

```suggestion
final readonly class InMemoryProvider implements ProviderInterface
{
    /**
     * @param array<string, (\Closure(): mixed)> $features
     */
    public function __construct(
        private array $features,
```